src/textual/widgets/_digit_display.py

```diff
@@ -0,0 +1,271 @@
+] = """
+···
+···
+···
+"""
+
+
+_character_map[
+    "X"
+] = """
+╻ ╻
+·╋·
+╹ ╹
+"""
+
+_character_map[
+    "Y"
+] = """
+╻ ╻
+·┳·
+·╹·
+"""
+
+_character_map[
+    "Z"
+] = """
+╺━┓
+·▞·
+┗━╸
+"""
+
+
+_character_map[
+    "."
+] = """
+··
+··
+·•
+"""
+
+
+_character_map[
+    ","
+] = """
+··
+··
+·▞
+"""
+
+_character_map[
+    "+"
+] = """
+···
+╺╋╸
+···
+"""
+
+_character_map[
+    "-"
+] = """
+···
+╺━╸
+···
+"""
+
+_character_map[
+    "="
+] = """
+···
+╺━·
+╺━·
+"""
+
+_character_map[
+    "*"
+] = """
+···
+·✱·
+···
+"""
+
+
+_character_map[
+    "/"
+] = """
+··╻
+·▞·
+╹··
+"""
+
+_character_map[
+    "^"
+] = """
+·╻·
+▝·▘
+···
+"""
+
+
+_VIRTUAL_SPACE = "·"
+
+# here we strip spaces and replace virtual spaces with spaces
+_character_map = {
+    k: v.strip().replace(_VIRTUAL_SPACE, " ") for k, v in _character_map.items()
+}
+
+
+class _SingleDigitDisplay(Static):
+    digit = reactive(" ", layout=True)
+
+    DEFAULT_CSS = """
+        _SingleDigitDisplay {
+          height: 3;
+          min-width: 2;
+          max-width: 3;
+        }
+    """
+
+    def __init__(self, initial_value=" ", **kwargs):
+        super().__init__(**kwargs)
+        self.digit = initial_value
+
+    def watch_digit(self, digit: str) -> None:
+        """Called when the digit attribute changes."""
+        if len(digit) > 1:
+            raise ValueError(f"Expected a single character, got {len(digit)}")
+        self.update(_character_map[digit.upper()])
+
+
+class DigitDisplay(Widget):
+    """A widget to display digits and basic arithmetic operators using Unicode blocks."""
+
+    digits = reactive("", layout=True)
+
+    supported_digits = frozenset(_character_map.keys())
+
+    DEFAULT_CSS = """
+    DigitDisplay {
+        layout: horizontal;
+        height: 3;
+    }
+    """
+
+    def __init__(self, initial_value="", **kwargs):
+        super().__init__(**kwargs)
+        self._displays = [_SingleDigitDisplay(d) for d in initial_value]
+        self.digits = initial_value
+
+    def compose(self) -> ComposeResult:
+        for widget in self._displays:
+            yield widget
+
+    def _add_digit_widget(self, digit: str) -> None:
+        new_widget = _SingleDigitDisplay(digit)
+        self._displays.append(new_widget)
+        self.mount(new_widget)
+
+    def watch_digits(self, digits: str) -> None:
+        """
+        Called when the digits attribute changes.
+        Here we update the display widgets to match the input digits.
+        """
+        diff_digits_len = len(digits) - len(self._displays)
+
+        # Here we add or remove widgets to match the number of digits
+        if diff_digits_len > 0:
+            start = len(self._displays)
+            for i in range(diff_digits_len):
+                self._add_digit_widget(digits[start + i])
+        elif diff_digits_len < 0:
+            for display in self._displays[diff_digits_len:]:
+                self._displays.remove(display)
+                display.remove()
```

Comment:
Tell me how you feel about this alternative in pseudocode

```py
while len(self._displays) < len(digits):
    self.add_digit_widget(digits[len(self._displays)])
while len(self._displays) > len(digits):
    self._displays.pop().remove()
```

This is much shorter and I tend to like this pattern, but I don't know how others react to it.

Reply:
Indeed, shorter and clearer, thanks!
I've done that, and inlined the `add_digit_wdiget` method, because I thought having the list manipulation here would make it more readable.

Reply:
Right.